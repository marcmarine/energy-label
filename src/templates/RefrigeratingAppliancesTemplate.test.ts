import { describe, it, expect } from 'vitest'
import { RefrigeratingAppliancesTemplate } from './RefrigeratingAppliancesTemplate'

describe('Regrigerating Appliances Template', () => {
  it('should generate complete arrow template', async () => {
    const template = new RefrigeratingAppliancesTemplate()
    const result = await template.generate({})

    expect(result).toMatchSnapshot()
  })
})
