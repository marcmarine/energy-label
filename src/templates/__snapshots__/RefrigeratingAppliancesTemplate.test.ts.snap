// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Regrigerating Appliances Template > should generate complete arrow template 1`] = `"<svg xmlns="http://www.w3.org/2000/svg" width="362.83" height="725.67" viewBox="0 0 362.83 725.67" fill="none"><rect width="362.83" height="725.67" fill="white" /><g transform="translate(11.34, 11.34) scale(0.79)"><path fill="#034EA2" d="M0 67h100V0H0v67Z"/><path fill="#FFF200" d="m48.88 10.94 1.03-3.36 1.02 3.36h3.48l-2.8 2.2 1.08 3.44-2.78-2.12-2.77 2.12 1.1-3.45-2.82-2.19h3.46ZM49.1 55.43l1.11-3.37 1.09 3.37h3.67l-2.96 2.19 1.15 3.44-2.95-2.11-2.96 2.11 1.17-3.44-3-2.19h3.67ZM60.11 52.62l1.06-3.38 1.02 3.38h3.45l-2.79 2.16 1.1 3.47-2.78-2.14-2.8 2.14 1.11-3.47-2.82-2.16h3.45ZM60.11 13.77l1.06-3.38 1.02 3.38h3.45l-2.79 2.17 1.1 3.46-2.78-2.13-2.8 2.13 1.11-3.46-2.82-2.17h3.45ZM68.54 22l1.04-3.16L70.6 22h3.47l-2.79 2.04 1.07 3.24-2.77-1.98-2.78 1.98 1.1-3.25L65.09 22h3.46ZM68.54 44.52l1.04-3.16 1.02 3.16h3.47l-2.79 2.04 1.07 3.24-2.77-2-2.78 2 1.1-3.24-2.82-2.04h3.46ZM71.33 33.25l1.06-3.15 1.02 3.15h3.47L74.1 35.3l1.07 3.24-2.77-2-2.78 2 1.09-3.24-2.8-2.05h3.43ZM37.64 13.75l1.06-3.36 1.01 3.36h3.46l-2.79 2.19 1.09 3.46-2.77-2.13-2.8 2.13 1.1-3.46-2.82-2.19h3.46ZM29.78 22.2l1.05-3.36 1.02 3.37h3.46l-2.8 2.18 1.1 3.45-2.78-2.12-2.79 2.13 1.1-3.46-2.82-2.18h3.46ZM26.97 33.26l1.05-3.16 1.02 3.16h3.46l-2.8 2.04 1.1 3.24-2.78-2-2.79 2 1.1-3.24-2.82-2.04h3.46ZM29.78 44.52l1.05-3.16 1.02 3.16h3.46l-2.8 2.04 1.1 3.24-2.78-2-2.79 2 1.1-3.24-2.82-2.04h3.46ZM37.86 52.6l1.1-3.36 1.1 3.37h3.67l-2.96 2.18 1.14 3.46-2.95-2.13-2.95 2.13 1.17-3.46-3-2.18h3.68Z"/></g><g transform="translate(98.27, 18.9) scale(0.91)"><path fill="#034EA2" d="M0 40.48V0h27.62v8.08H8.42v7.73h18.4v7.9H8.42v8.86h19.2v7.9H0ZM58.39 27.26V0h8.28v40.48h-9.89l-16.6-26.91v26.9H31.9V0h9.88l16.6 27.26ZM71.43 40.48V0h27.14v8.08H79.71v7.73h18.1v7.9H79.7v8.86h18.86v7.9H71.43ZM102.86 40.48V0h19.17c1.93-.03 3.86.19 5.73.66a9.22 9.22 0 0 1 3.95 2.14 8.89 8.89 0 0 1 2.3 3.92c.54 1.93.79 3.94.75 5.94 0 1.23-.08 2.46-.25 3.68-.13 1.06-.35 2.11-.65 3.14a6.1 6.1 0 0 1-1.92 2.7c.8.87 1.4 1.9 1.76 3.03.39 1.25.58 2.55.57 3.86v11.4h-8.59V29.99a3.79 3.79 0 0 0-1.07-3.13 5.55 5.55 0 0 0-3.42-.84h-10.02v14.47h-8.31Zm8.38-22.33h9.4c.99.02 1.97-.06 2.94-.24.6-.1 1.16-.31 1.67-.61.6-.49.9-1.71.9-3.68.1-1.5-.08-2.99-.54-4.4-.36-.8-1.53-1.22-3.53-1.22h-10.84v10.15ZM153.39 18.36h19.6c.2 1.53.3 2.67.33 3.4.03.75 0 1.43 0 1.97v3.32c.05 2.83-.4 5.64-1.32 8.32a7.14 7.14 0 0 1-2.82 3.59 12.42 12.42 0 0 1-5.34 1.72 65.43 65.43 0 0 1-7.6.24c-6.44 0-10.86-1.02-13.27-3.07-2.62-2.13-3.92-6.33-3.92-12.6V15.4c0-5.22 1.06-9.08 3.2-11.58a10.22 10.22 0 0 1 4.6-2.85 22.18 22.18 0 0 1 6.91-.97h8.68c3.94.13 6.7 1.18 8.29 3.17 1.8 2.46 2.69 5.46 2.53 8.5v3.53h-8.44c.16-2.13-.06-4.27-.67-6.32-.2-.4-1.03-.64-2.53-.73a81.8 81.8 0 0 0-3.65-.14h-3.7c-3.1 0-4.93.32-5.5.97-.56.64-.88 2.87-.96 6.7v9.76a24 24 0 0 0 .53 5.49v-.12c.22.47.55.87.97 1.18.57.4 1.24.64 1.93.7.82.12 1.65.18 2.48.19H158.75c1.14 0 2.28-.04 3.41-.16a3 3 0 0 0 1.78-.63c.56-.48.85-2.1.85-4.81v-.97h-11.4v-7.95ZM193.68 0l-3.65 11.67 9.97-1.66-14.26 31.42 4.3-22.77-9.56.41L184.52 0h9.16Z"/></g><image href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAIACAYAAAD0eNT6AAAAAklEQVR4AewaftIAAAwXSURBVO3BQRblCG4ksCBf3v/KMb2v2thSTsufACZJwxlt84aZya9pm6dmJl/SNk/NTN7QNl8xM/k1bfOGmQk3bACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBz/uQlbcPfMzPh+9rmDTOTr2ibL5mZPNU2XzIz+TVtw98zM3lqAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJzzJx8zM/k1bfNr2oZ/mpm8oW34d23z1MzkS9qGf5qZ/Jq2+YoNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnD+B/4WZyVe0Df9uZvIVbfOGmclXtM0bZiZPtQ38T2wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAc/4E/hfa5itmJm9oG/6emckb2uYrZiZvaBv4/20DAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnPMnH9M23DAz+ZKZyVNt84aZyRva5tfMTH7NzOSptvk1bcPfswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJzzJy+ZmXDHzOSptnnDzOQNbfPUzOQNbfOGmclTbfOGmckb2uapmckb2uYNM5NfMzPh2zYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAOX/aBv4bZib8XW3za2YmT7XNG2Ymb2ibr2gbbtgAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA50yS5gVt84aZyVe0zRtmJr+mbfinmcmXtM1XzEx+Tdv8mpnJG9rmK2Ymb2ibp2YmX7EBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcM/2PvGBm8hVt84aZyVe0DXfMTL6ibd4wM3lD2zw1M+GOtvk1M5OnNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5f/IxbfPUzORL2uapmcmvaZtfMzP5krb5irZ5w8zkK9rmDTOTX9M2v2Zm8hVt89QGADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGeSNC9omzfMTL6ibd4wM3mqbb5kZsLf0zZfMTN5Q9t8xczkDW3za2YmX9E2/LuZyVMbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAOdP/CP9qZvJr2uYNMxP+nrZ5w8zkqbb5kpnJU23zJTOTr2ibXzMz+Yq2+YoNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM70P/KCmckb2uapmcmXtM1XzEze0DZPzUx+Tdt8ycyE72ubr5iZvKFtfs3M5JdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHP+5GNmJk+1zRtmJl8xM3lD2/D3zEy+pG34p5nJl8xMvqJtvmJm8oa24Z82AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAc6b/EfgvmJl8Sdv8mpnJV7QN/25m8mva5qmZyRva5itmJm9om6c2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDl/ZiZvaJuvmJm8oW2+YmbyJW3zVNv8mpnJG9qGO9rmqZnJG9rm18xM3tA2v2QDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnPOnbX5N23zJzOSptvmSmclTbfOGmQl3zEyeapsvmZk81TZfMjN5qm3eMDN5w8zkl2wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADn/JmZvKFt3jAz+Yq2eUPbPDUz+ZK2+Yq2ecPM5Km2ecPMBP4va5unZiZvaJs3zEyeapuv2AAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADn/Gkb/q6ZyVNt84aZyVfMTN7QNm9oG/6emclXzEze0DZvaJunZiZvaJs3zEyeahv+ng0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzvQ/8oKZCf+ubZ6amXxJ2/yamQn/1DZvmJnAdW3zFRsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA50/8I/A/NTL6ibfi+mckb2ubXzEyeaptfMzPh79kAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA5/yZmXBH23xF27xhZvJr2uYNM5On2uZLZib8PTOTN7TNr2mbp2Ymb2ibpzYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAOX/ykrbh75mZ/JqZyZe0zVfMTH5N23zFzOQNbfOGmclXtA1/T9u8YWby1AYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOn3zMzOTXtM2vaZunZiZfMjP5irb5ipnJG9rmDTOTp9rm18xM+Hdt84aZyVNt84a2eWoDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnPMn8L8wM/mKtnnDzOSptnnDzOQNbfMVM5M3tM2vaZunZiZvaJuvmJm8YWbyhrb5ipnJUxsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnD+B/5K2ecPM5A1t89TM5A1t84aZyVNt84aZyRtmJl/RNm+YmTzVNl8yM3mqbb5kZvJU23zFBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5GwDgnA0AcM6ffEzb8H1t89TM5A1t82tmJm9om6dmJl/SNk/NTH7NzOQNbfMVM5MvaZunZiZvaJunNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCcDQBwzgYAOGcDAJyzAQDO2QAA52wAgHM2AMA5f/KSmQl3zEyeaps3zEze0DZPtc2vaRvumJm8oW34tg0AcM4GADhnAwCcswEAztkAAOdsAIBzNgDAORsA4JwNAHDOBgA4ZwMAnLMBAM7ZAADnbACAczYAwDkbAOCc/wcm9TIRrzHGnwAAAABJRU5ErkJggg==" x="298.58" y="11.34" width="52.91" /><text id="supplier-name" fill="black" font-family="Verdana" font-size="12" font-weight="bold"><tspan x="11.34" y="94.49">Supplier's Name</tspan></text><text id="model-identifier" fill="black" font-family="Verdana" font-size="12" text-anchor="end"><tspan x="351.5" y="94.49">Model Identifier</tspan></text><path d="M11.34 102.05H351.5" stroke="black" stroke-width="0.5pt" /><g transform="translate(11.34, 117.17)"><path d="M0,0 H71.81 L86.93,16.06 L71.81,32.13 H0 Z" fill="#00A651" /><text x="11.34" y="37.8" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">A</text><g transform="translate(0,-6.61)"><g transform="translate(0,0)"><polygon points="253.23,22.68 283.46,0 340.16,0 340.16,45.35 283.46,45.35" fill="black" /><text x="298.58" y="35.91" font-size="33pt" font-weight="bold" fill="white" font-family="Calibri">A</text></g></g><path d="M0,37.8 H94.49 L109.61,53.86 L94.49,69.93 H0 Z" fill="#50B848" /><text x="11.34" y="75.6" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">B</text><path d="M0,75.6 H109.61 L124.73,91.66 L109.61,107.72999999999999 H0 Z" fill="#BFD730" /><text x="11.34" y="113.39999999999999" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">C</text><path d="M0,113.39999999999999 H120.94 L136.06,129.45999999999998 L120.94,145.53 H0 Z" fill="#FFF200" /><text x="11.34" y="151.2" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">D</text><path d="M0,151.2 H136.06 L151.18,167.26 L136.06,183.32999999999998 H0 Z" fill="#FDB913" /><text x="11.34" y="189" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">E</text><path d="M0,189 H151.18 L166.3,205.06 L151.18,221.13 H0 Z" fill="#F37021" /><text x="11.34" y="226.8" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">F</text><path d="M0,226.79999999999998 H166.3 L181.42000000000002,242.85999999999999 L166.3,258.93 H0 Z" fill="#ED1C24" /><text x="11.34" y="264.59999999999997" transform="translate(0, -14)" font-family="Calibri" font-size="19pt" font-weight="bold" fill="white">G</text></g><path d="M11.34 391.18H351.5" stroke="black" stroke-width="0.5pt" /><text x="181.42" y="444.09" fill="black" font-family="Verdana" text-anchor="middle"><tspan font-size="28pt" font-weight="bold">XYZ</tspan><tspan font-size="18pt">kWh/annum</tspan></text><path d="M11.34 466.77H351.5" stroke="black" stroke-width="0.5pt" /><g transform="translate(73.7, 504.57)"><path d="M23.675 1V53.91" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M15.9785 4.93251L23.5507 12.7837L31.3854 4.67126" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M31.3854 49.3725L23.7994 41.5212L15.9785 49.6337" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M1.80139 14.3788L46.0598 40.82" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M1.22107 23.2612L11.543 20.3875L8.68267 9.27751" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M46.1289 31.6487L35.807 34.5225L38.6673 45.6325" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M1.80139 40.655L46.0598 14.2138" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M8.90377 45.6325L11.6673 34.88L1 31.91" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/><path d="M38.4462 9.53875L35.6826 20.2912L46.35 23.2612" stroke="#231F20" stroke-width="1.59792" stroke-linecap="round" stroke-linejoin="round"/></g><text x="96.38" y="583.94" fill="black" font-family="Verdana" text-anchor="middle"><tspan font-size="16pt" font-weight="bold" >XYZ</tspan><tspan font-size="12pt"> L</tspan></text><g transform="translate(253.23, 504.57)"><path d="M4.31961 15.675H24.1266M3.94615 7.87679L1 15.4V53.91H27.46V15.1799L24.5138 7.87679H3.94615ZM4.31961 1V7.87679H24.1266V1H4.31961Z" stroke="#231F20" stroke-width="1.59792" stroke-linejoin="round"/></g><text x="266.46" y="583.94" fill="black" font-family="Verdana" text-anchor="middle"><tspan font-size="16pt" font-weight="bold" >XYZ</tspan><tspan font-size="12pt"> L</tspan></text><g transform="translate(141.73, 610.39)"><path d="M68.6381 45.8273C70.9479 40.3119 72.0396 34.3628 71.8383 28.3866C71.6371 22.4105 70.1477 16.5483 67.4721 11.2007M77.1015 49.5088C79.9471 42.735 81.2928 35.4256 81.0463 28.0825C80.7998 20.7394 78.9669 13.5367 75.6735 6.96903M59.952 42.7223C61.8093 38.2791 62.6876 33.4878 62.5271 28.6747C62.3666 23.8615 61.171 19.1394 59.0218 14.8298M49.3531 0.876961V58.7189L26.6879 44.8578H1.07501V15.0394H26.6748L49.34 0.876961H49.3531Z" stroke="black" stroke-width="1.2pt" stroke-linecap="round" stroke-linejoin="round"/></g><text x="186.33" y="646.3" fill="black" font-family="Verdana" text-anchor="end"><tspan font-size="12pt" font-weight="bold">XY</tspan><tspan font-size="9pt">dB</tspan></text><text x="181.42" y="689.76" fill="black" font-family="Verdana" text-anchor="middle"><tspan font-size="16pt" font-weight="bold">A</tspan><tspan font-size="10pt" font-weight="normal">B</tspan><tspan font-size="10pt" font-weight="normal">C</tspan><tspan font-size="10pt" font-weight="normal">D</tspan></text><text transform="translate(343.94 714.33) rotate(90)" fill="black" font-family="Verdana" font-size="6pt" text-anchor="end"> 2019/2016</text></svg>"`;
